src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -423,10 +439,66 @@ private ContextMenu updateContextMenu(ContextMenu contextMenu) {
         updateMarkAsReadUnreadMenuItem();
         updateChangeLabelsMenuItem();
         updateChangeMilestoneMenuItem();
+        updateAvailableWatchListPanels();
 
         return contextMenu;
     }
 
+    private void updateAvailableWatchListPanels() {
+        restoreDefaultAddToWatchListPanelMenu();
+        Optional<GuiElement> item = listView.getSelectedItem();
+
+        if (item.isPresent()) {
+            addToWatchListMenu.setDisable(false);
+        } else {
+            addToWatchListMenu.setDisable(true);
+        }
+
+        parentPanelControl.getChildren().forEach(node -> {
+            if (isWatchListPanel(node)) {
+                MenuItem currentWatchPanel = createWatchListPanelMenuItem((FilterPanel) node);
+                addToWatchListMenu.getItems().add(currentWatchPanel);
+            }
+        });
+    }
+
+    private void restoreDefaultAddToWatchListPanelMenu() {
+        addToWatchListMenu.getItems().clear();
+        addToWatchListMenu.getItems().add(newWatchListMenuItem);
+    }
+
+    /**
+     * Creates a watch panel menu item for an existing watch list panel so that
+     * the user can add an issue to that panel
+     */
+    private MenuItem createWatchListPanelMenuItem(FilterPanel watchListPanel) {
+        MenuItem watchPanelMenu = new MenuItem();
+        String watchListPanelTitle = watchListPanel.getCurrentInfo().getPanelName();
+        watchPanelMenu.setText(watchListPanelTitle);
+        watchPanelMenu.setOnAction(e -> addSelectedIssueToWatchListPanel(watchListPanel));
+        return watchPanelMenu;
+    }
+
+    /**
+     * A (filter) panel is a watch list panel if its qualifier is empty or its qualifier only has id qualifier.
```

Comment:
I think this is enough? Otherwise you end up defining watch list filter in multiple places.
`@return true if the panel's filter is a 'watch list filter'`